### PR TITLE
Move shared variables in conf.yaml to the top

### PR DESCRIPTION
**Pull request by gtback**
I also added a script that's used to test changes to ensure YAML refactors don't change the interpreted meaning. I used the script when making this change to ensure that the build would interpret conf.yaml the same way before and after this change.

### CODE REVIEW

**Comment by gtback** [comment]
Once this is merged, I'll update:

- [ ] #2184
- [ ] #2203

As well as fixing any PRs with conflicts or new PRs that are trying to use the `mapMasterToMain` and `mapMainToMaster` but needing to move the definitions around.

File: conf.yaml [hunk 1]
```diff
@@ -1345,7 +1369,7 @@ contents:
                   -
                     repo:   ecs-logging
                     path:   docs
-                    map_branches: &mapEcsLoggingJavaToEcsLogging
```
Comment (bmorelli25) [inline conf.yaml:1348]: Why did this one disappear instead of moving to the top?

Comment (lcawl) [reply conf.yaml:1348]: I assume because it's only used once and therefore the attribute isn't necessary?

Comment (bmorelli25) [reply conf.yaml:1348]: Ah, that makes sense. Thanks!

File: conf.yaml [hunk 2]
```diff
@@ -2397,7 +2421,6 @@ contents:
             current:    6.2
             subject:    X-Pack
             index:      docs/en/index.asciidoc
-            private:    1
```
Comment (lcawl) [inline conf.yaml:2400]: > private: 1

I suspect these "private" settings were used to omit "edit URLs" from these legacy docs.  Is this change necessary?

File: conf.yaml (hunk 1 above)
Comment (gtback) [reply conf.yaml:1348]: Yep, there were a couple that were only used once, so I just removed them. I thought about including them anyway, but then realized there was no reason to not make all of the lists into variables.

File: conf.yaml (hunk 2 above)
Comment (gtback) [reply conf.yaml:2400]: `private` is defined twice for these books (see line 2418 in the new version or 2394 in the old version). I've been meaning to clean this up forever (VS Code reports the problem when I'm editing the conf.yaml file)

**Review by szabosteve — APPROVED** [review]
LGTM! Thanks! 🚀

**Review by lcawl — APPROVED** [review]
LGTM2, thanks!
